### PR TITLE
Try using the IsolatedBlockEditor

### PR DESCRIPTION
**Pull request by sarayourfriend**
#### Changes proposed in this Pull Request

* This is a WIP branch to try to get the IsolatedBlockEditor working (relatively) for reader comments

#### Testing instructions

*

Fixes #

### CODE REVIEW

**Comment by saramarcondes** [comment]
@johngodley if you have a few minutes, would you be able to take a look at this and help me figure out what's missing from this (other than obvious things like actually consuming the blocks rather than logging them).

File: client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/editor-menu.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+	return (
+		<MenuGroup label={ _x( 'Editor', 'noun' ) }>
+			<MenuItem
+				icon={ editorMode === 'visual' && check }
+				isSelected={ editorMode === 'visual' }
+				onClick={ () => setMode( 'visual' ) }
+				role="menuitemcheckbox"
+			>
+				{ __( 'Visual editor' ) }
```
Comment (a8ci18n) [inline client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/editor-menu.js:55]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **123** times:
`translate( 'Visual Editor' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/try/isolated-block-editor)

File: client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/editor-menu.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			</MenuItem>
+
+			<MenuItem
+				icon={ editorMode === 'text' && check }
+				isSelected={ editorMode === 'text' }
+				onClick={ () => setMode( 'text' ) }
+				role="menuitemcheckbox"
+			>
+				{ __( 'Code editor' ) }
```
Comment (a8ci18n) [inline client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/editor-menu.js:64]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **93** times:
`translate( 'Code Editor' )` *ES Score: 9*

File: client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ * @param {object} props - Component props
+ * @param {BlockEditorSettings} props.settings - Settings
+ * @param {OnClose} props.onClick
+ * @param {OnMore} props.renderMoreMenu
+ */
+const MoreMenu = ( { settings, onClick, renderMoreMenu } ) => (
+	<DropdownMenu
+		className="edit-post-more-menu"
+		icon={ moreVertical }
+		label={ __( 'More tools & options' ) }
```
Comment (a8ci18n) [inline client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/index.js:55]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **83** times:
`translate( 'Hide more tools & options' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/try/isolated-block-editor)

File: client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/writing-menu.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		return null;
+	}
+
+	return (
+		<MenuGroup label={ _x( 'View', 'noun' ) }>
+			{ topToolbar && (
+				<>
+					<FeatureToggle
+						feature="fixedToolbar"
+						label={ __( 'Top toolbar' ) }
```
Comment (a8ci18n) [inline client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/writing-menu.js:49]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **106** times:
`translate( 'Top Toolbar' )` *ES Score: 11*

File: client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/writing-menu.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+						info={ __( 'Access all block and document tools in a single place.' ) }
```
Comment (a8ci18n) [inline client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/writing-menu.js:50]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **95** times:
`translate( 'Access all block and document tools in a single place' )` *ES Score: 8*

File: client/components/isolated-editor/client/components/block-editor/convert-to-group-buttons.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+			{ ( { onClose } ) => (
+				<>
+					{ isGroupable && (
+						<MenuItem
+							onClick={ () => {
+								onConvertToGroup();
+								onClose();
+							} }
+						>
+							{ _x( 'Group', 'verb' ) }
```
Comment (a8ci18n) [inline client/components/isolated-editor/client/components/block-editor/convert-to-group-buttons.js:37]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **33** times:
`translate( 'Group', { context: 'NewDash Page Title'} )` *ES Score: 11*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/try/isolated-block-editor)


:information_source: This string already exists with the following **contexts**:
- `null` (no context)
- `NewDash Page Title`

 Would it make sense to reuse one of the above?

File: client/components/isolated-editor/client/components/block-editor/convert-to-group-buttons.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+					) }
+					{ isUngroupable && (
+						<MenuItem
+							onClick={ () => {
+								onConvertFromGroup();
+								onClose();
+							} }
+						>
+							{ _x(
+								'Ungroup',
```
Comment (a8ci18n) [inline client/components/isolated-editor/client/components/block-editor/convert-to-group-buttons.js:48]: :information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/components/isolated-editor/client/components/block-editor-toolbar/header-toolbar/index.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+							if ( isInserterOpened ) {
+								// Focusing the inserter button closes the inserter popover
+								inserterButton.current.focus();
+							} else {
+								setIsInserterOpened( true );
+							}
+						} }
+						disabled={ ! isInserterEnabled }
+						icon={ plus }
+						label={ _x( 'Add block', 'Generic label for block inserter button' ) }
```
Comment (a8ci18n) [inline client/components/isolated-editor/client/components/block-editor-toolbar/header-toolbar/index.js:90]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **100** times:
`translate( 'Add block' )` *ES Score: 10*


:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/editor-menu.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+		onSetMode( mode );
+		onClose();
+	};
+
+	if ( ! isCodeEditingEnabled || ! settings.iso.moreMenu.editor ) {
+		return null;
+	}
+
+	return (
+		<MenuGroup label={ _x( 'Editor', 'noun' ) }>
```
Comment (a8ci18n) [inline client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/editor-menu.js:48]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **37** times:
`translate( 'Editor', { context: 'Noun: A user role displayed in a badge'} )` *ES Score: 13*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/try/isolated-block-editor)


:information_source: This string already exists with the following **contexts**:
- `null` (no context)
- `Noun: A user role displayed in a badge`
- `User role`

 Would it make sense to reuse one of the above?

File: client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/writing-menu.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+		[]
+	);
+
+	// Anything to show?
+	if ( ! fullscreen && ! preview && ! topToolbar ) {
+		return null;
+	}
+
+	return (
+		<MenuGroup label={ _x( 'View', 'noun' ) }>
```
Comment (a8ci18n) [inline client/components/isolated-editor/client/components/block-editor-toolbar/more-menu/writing-menu.js:44]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **38** times:
`translate( 'View', 'Views', { context: 'noun'} )` *ES Score: 13*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/try/isolated-block-editor)


:information_source: This string already exists with the following **contexts**:
- `null` (no context)
- `Stats: List item action to view content`
- `TinyMCE menu`
- `verb`
- `view a subscription`

 Would it make sense to reuse one of the above?

**Comment by johngodley** [comment]
There are two main issues at the moment:
- Setup. The `IsolatedBlockEditor` component needs settings as per the readme in the PR. The component itself would look something like this:

```js
render(
	<IsolatedBlockEditor
		settings={ settings }
		onSaveContent={ saveContent }
		onLoad={ ( parser ) => parser( contentHTML ) }
		onError={ () => document.location.reload() }
	/>,
	document.querySelector( '#container' )
);
```

- Gutenberg. The component doesn't load Gutenberg itself. On P2 this is loaded separately as part of WordPress. In the other experiments I tried different combinations of bundling Gutenberg with the code, and loading Gutenberg manually. I haven't settled on anything yet, and the needs of Calypso may influence this.

I know very little about the reader (or Calypso, for that matter), so the loading of Gutenberg is mostly questions:

- Where would Gutenberg load from? Is the code available on the wordpress.com domain and can we reference it? Would you need your own version?
- Do you want to fix yourself to a particular version, or update as and when wordpress.com updates? The later means you're always using the latest code, but opens you up to potential problems if a Gutenberg change breaks something
- Does the editor need any site-specific settings, similar to the full editor (editor styles, list of blocks)?

**Comment by saramarcondes** [comment]
Thanks for the help @johngodley!

> component needs settings as per the readme in the PR

Okay! I was relying on the default settings but I think I must have misunderstood how the editor was using those. I'll take another look at the settings.

> Where would Gutenberg load from? Is the code available on the wordpress.com domain and can we reference it? Would you need your own version?

Gutenberg does not currently exist inside of Calypso (sort of... it does exist for Gutenboarding) so I think we'd need to include it ourselves or have it bundled as part of the isolated block editor. Think of Calypso was a fully non-WordPress context with access to the WordPress.com API. That's what we have (so probably similar to how you integrated it into that chat application, that's what Calypso is going to need to look like I would think).

> Do you want to fix yourself to a particular version, or update as and when wordpress.com updates?

I think preferrably we're managing the GB version independent of the WP.com block editor. In my mind we'd be consuming the Gutenberg packages directly rather than through the plugin or the core version. Does that make sense?

> Does the editor need any site-specific settings, similar to the full editor (editor styles, list of blocks)?

I'm not sure about editor styles (not actually familiar with what you mean by that) but the reader comments would use a specific set of blocks, not the full block list for any particular WP.com site.

/cc @griffbrad who may be able to help answer these questions better than I can.

**Pull request closed without merging**